core/src/test/scala/org/apache/spark/sql/expression/index/CoveringIndex0Suite.scala

```diff
@@ -0,0 +1,499 @@
+    "select tp_char from full_data_type_table_idx where tp_char < 'f5ydp4uwoA' and tp_int != 2333",
+    "select tp_char from full_data_type_table_idx where tp_int < 2333 and tp_char > 'f5ydp4uwoA'",
+    "select tp_char from full_data_type_table_idx where tp_char > 'f5ydp4uwoA' and tp_int != 2333",
+    "select tp_char from full_data_type_table_idx where tp_int < 2333 and tp_char <= 'f5ydp4uwoA'",
+    "select tp_char from full_data_type_table_idx where tp_char <= 'f5ydp4uwoA' and tp_int != 2333",
+    "select tp_char from full_data_type_table_idx where tp_int < 2333 and tp_char >= 'f5ydp4uwoA'",
+    "select tp_char from full_data_type_table_idx where tp_char >= 'f5ydp4uwoA' and tp_int != 2333",
+    "select tp_char from full_data_type_table_idx where tp_int < 2333 and tp_char = 'f5ydp4uwoA'",
+    "select tp_char from full_data_type_table_idx where tp_char = 'f5ydp4uwoA' and tp_int != 2333",
+    "select tp_char from full_data_type_table_idx where tp_int < 2333 and tp_char != 'f5ydp4uwoA'",
+    "select tp_char from full_data_type_table_idx where tp_char != 'f5ydp4uwoA' and tp_int != 2333",
+    "select tp_char from full_data_type_table_idx where tp_int < 2333 and tp_char <> 'f5ydp4uwoA'",
+    "select tp_char from full_data_type_table_idx where tp_char <> 'f5ydp4uwoA' and tp_int != 2333",
+    "select tp_nvarchar from full_data_type_table_idx where tp_int < 2333 and tp_nvarchar < '5vNT8'",
+    "select tp_nvarchar from full_data_type_table_idx where tp_nvarchar < '5vNT8' and tp_int != 2333",
+    "select tp_nvarchar from full_data_type_table_idx where tp_int < 2333 and tp_nvarchar > '5vNT8'",
+    "select tp_nvarchar from full_data_type_table_idx where tp_nvarchar > '5vNT8' and tp_int != 2333",
+    "select tp_nvarchar from full_data_type_table_idx where tp_int < 2333 and tp_nvarchar <= '5vNT8'",
+    "select tp_nvarchar from full_data_type_table_idx where tp_nvarchar <= '5vNT8' and tp_int != 2333",
+    "select tp_nvarchar from full_data_type_table_idx where tp_int < 2333 and tp_nvarchar >= '5vNT8'",
+    "select tp_nvarchar from full_data_type_table_idx where tp_nvarchar >= '5vNT8' and tp_int != 2333",
+    "select tp_nvarchar from full_data_type_table_idx where tp_int < 2333 and tp_nvarchar = '5vNT8'",
+    "select tp_nvarchar from full_data_type_table_idx where tp_nvarchar = '5vNT8' and tp_int != 2333",
+    "select tp_nvarchar from full_data_type_table_idx where tp_int < 2333 and tp_nvarchar != '5vNT8'",
+    "select tp_nvarchar from full_data_type_table_idx where tp_nvarchar != '5vNT8' and tp_int != 2333",
+    "select tp_nvarchar from full_data_type_table_idx where tp_int < 2333 and tp_nvarchar <> '5vNT8'",
+    "select tp_nvarchar from full_data_type_table_idx where tp_nvarchar <> '5vNT8' and tp_int != 2333",
+    "select tp_int from full_data_type_table_idx where tp_real > 0.7 and tp_int >= 2333",
+    // cover index tests: select {S} where {T} (S ∩ T != ∅)
```

Comment:
mark the line where complex tests are generated.

Reply:
Do we have cases that covers prefix index?

Reply:
Yes, all columns like `varchar(45)` with specificate length can be used as prefix index.